### ISSUE TITLE
Pipeline names are not propagated in logs

### ISSUE THREAD

**Issue by krisztianfekete** [open-telemetry/opentelemetry-collector#7033]
### Component(s)

receiver/prometheus

## Description

Currently only the component identifier is logged, so if I have two pipelines, e.g. `metrics/target1` and `metrics/target2`, I won't be able to see which pipeline is misconfigured.

## Steps to Reproduce

Sample pipeline config:
```yaml
      pipelines:
        metrics:
          receivers:
          - otlp
          - prometheus
          processors:
          - memory_limiter
          - batch
          exporters:
          - prometheus
        metrics/target2:
          exporters:
          - prometheus
          processors:
          - memory_limiter
          - batch
          receivers:
          - prometheus/target2
```

Misconfigure the `prometheus/target2` receiver, e.g.

```yaml
    prometheus/gloo-mgmt:
      config:
        scrape_configs:
        - job_name: target2-otel
          honor_labels: true
          kubernetes_sd_configs:
          - namespaces:
              names:
              - default
            role: pod
          relabel_configs:
          - action: keep
            regex: target2|target2-ui
            source_labels:
            - __meta_kubernetes_pod_label_app
          - action: keep
            regex: true
            source_labels:
            - __meta_kubernetes_pod_annotation_prometheus_io_scrape
          - action: drop
            regex: true
            source_labels:
            - __meta_kubernetes_pod_annotation_prometheus_io_scrape_slow
          - action: replace
            regex: (https?)
            source_labels:
            - __meta_kubernetes_pod_annotation_prometheus_io_scheme
            target_label: __scheme__
          - action: replace
            regex: (.+)
            source_labels:
            - __meta_kubernetes_pod_annotation_prometheus_io_path
            target_label: __metrics_path__
          - action: replace
            regex: ([^:]+)(?::\d+)?;(\d+)
            replacement: $1:$2
            source_labels:
            - __address__
            - __meta_kubernetes_pod_annotation_prometheus_io_port
            target_label: __address__
          - action: labelmap
            regex: __meta_kubernetes_pod_annotation_prometheus_io_param_(.+)
            replacement: __param_$1
          - action: labelmap
            regex: __meta_kubernetes_pod_label_(.+)
          - action: replace
            source_labels:
            - __meta_kubernetes_namespace
            target_label: namespace
          - action: replace
            source_labels:
            - __meta_kubernetes_pod_name
            target_label: pod
          - action: drop
            regex: Pending|Succeeded|Failed|Completed
            source_labels:
            - __meta_kubernetes_pod_phase
```

Notice that I am using only single `$` chars, instead of `$$`, so this will yield the error seen below.

## Expected Result

In the logs, I should see the actual name of the pipeline that is misconfigured, something like this:

```console
2023-01-17T15:48:56.089Z	warn	internal/transaction.go:121	Failed to scrape Prometheus endpoint	{"kind": "receiver", "name": "prometheus/target2", "pipeline": "metrics/target2", "scrape_timestamp": 1673970536089, "target_labels": "{__name__=\"up\", app=\"target2\", instance=\":\", job=\"target2-otel\", namespace=\"target2\", pod=\"target2-7888ff5588-md59m\", pod_template_hash=\"7888ff5588\"}"}
2023-01-17T15:49:49.049Z	warn	internal/transaction.go:121	Failed to scrape Prometheus endpoint	{"kind": "receiver", "name": "prometheus/target2", "pipeline": "metrics/target2", "scrape_timestamp": 1673970589047, "target_labels": "{__name__=\"up\", app=\"target2-ui\", target2=\"target2-ui\", instance=\":\", job=\"target2-otel\", namespace=\"target2\", pod=\"target2-ui-75c487f979-9nc8z\", pod_template_hash=\"75c487f979\"}"}
```

## Actual Result

I can only see metrics component identifier:

```console
2023-01-17T15:48:56.089Z	warn	internal/transaction.go:121	Failed to scrape Prometheus endpoint	{"kind": "receiver", "name": "prometheus/target2", "pipeline": "metrics", "scrape_timestamp": 1673970536089, "target_labels": "{__name__=\"up\", app=\"target2\", instance=\":\", job=\"target2-otel\", namespace=\"target2\", pod=\"target2-7888ff5588-md59m\", pod_template_hash=\"7888ff5588\"}"}
2023-01-17T15:49:49.049Z	warn	internal/transaction.go:121	Failed to scrape Prometheus endpoint	{"kind": "receiver", "name": "prometheus/target2", "pipeline": "metrics", "scrape_timestamp": 1673970589047, "target_labels": "{__name__=\"up\", app=\"target2-ui\", target2=\"target2-ui\", instance=\":\", job=\"target2-otel\", namespace=\"target2\", pod=\"target2-ui-75c487f979-9nc8z\", pod_template_hash=\"75c487f979\"}"}
```

### Collector version

0.67.0

## Environment
Irrelevant

### OpenTelemetry Collector configuration

```yaml
pipelines:
        metrics:
          receivers:
          - otlp
          - prometheus
          processors:
          - memory_limiter
          - batch
          exporters:
          - prometheus
        metrics/target2:
          exporters:
          - prometheus
          processors:
          - memory_limiter
          - batch
          receivers:
          - prometheus/target2
```

### Log output

```shell
2023-01-17T15:48:56.089Z	warn	internal/transaction.go:121	Failed to scrape Prometheus endpoint	{"kind": "receiver", "name": "prometheus/target2", "pipeline": "metrics", "scrape_timestamp": 1673970536089, "target_labels": "{__name__=\"up\", app=\"target2\", instance=\":\", job=\"target2-otel\", namespace=\"target2\", pod=\"target2-7888ff5588-md59m\", pod_template_hash=\"7888ff5588\"}"}
2023-01-17T15:49:49.049Z	warn	internal/transaction.go:121	Failed to scrape Prometheus endpoint	{"kind": "receiver", "name": "prometheus/target2", "pipeline": "metrics", "scrape_timestamp": 1673970589047, "target_labels": "{__name__=\"up\", app=\"target2-ui\", target2=\"target2-ui\", instance=\":\", job=\"target2-otel\", namespace=\"target2\", pod=\"target2-ui-75c487f979-9nc8z\", pod_template_hash=\"75c487f979\"}"}
```

### Additional context

https://cloud-native.slack.com/archives/C01N6P7KR6W/p1674213181401129

**Comment by Aneurysm9** [comment]
I've moved this to the core collector repo as this is not an issue with the receiver but with the creation of a logger for receivers which happens in the collector service when it constructs pipelines.  The relevant code is [here](https://github.com/open-telemetry/opentelemetry-collector/blob/12c782d9c7883759246bd516bb202e753bd5b202/service/internal/components/loggers.go#L41) and, unfortunately, I'm not sure that it is possible to do what this issue asks.  As receivers can be part of multiple pipelines with a single receiver instance the configuration provided to that receiver instance cannot include information about which pipeline it is a part of.  You do, however, get the full component ID for the receiver, including the name, which can be used to identify the configuration at issue.

**Comment by krisztianfekete** [comment]
Hey @Aneurysm9, thanks for taking a look and moving the issue to its correct place. I see the limitations that you mentioned. I am wondering if there are ways to improve logging without adding the actual name of the pipeline in use.

I had another metrics pipeline called `metrics`, and when I checked the logs I was confused and thought that somehow my pipelines could have been merged, as the other one was listed in the messages, and I expected that to reflect the actual pipeline in use.

Yes, the component ID of the receiver is there, but in my particular case that partially led to the confusion as I could not image how my receiver in `metrics/target2` could have affect the pipeline called `metrics`.

What I did was setting the log level to DEBUG, and changing the name of the other pipeline to something else, and that was the moment I realised that it's not the actual name of the affected pipeline.

Just thinking out loud: maybe the name of the pipeline is more confusing here than it is helpful actually? Since as you also mentioned, the name of the receiver should identify the source of the problem.
